src/Panel/Panel.php

```diff
@@ -45,7 +200,290 @@ public static function customCss(App $kirby)
+            '$multilang' => $multilang,
+            '$url' => Url::current(),
+            '$user' => function () use ($kirby) {
+                if ($user = $kirby->user()) {
+                    return [
+                        'email'       => $user->email(),
+                        'id'          => $user->id(),
+                        'language'    => $user->language(),
+                        'role'        => $user->role()->id(),
+                        'username'    => $user->username(),
+                    ];
+                }
+
+                return null;
+            }
+        ];
+
+        // merge with shared data
+        return array_merge($shared, $data);
+    }
+
+    /**
+     * Renders the error view with provided message
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @param string $message
+     * @return \Kirby\Http\Response
+     */
+    public static function error(App $kirby, string $message)
+    {
+        return [
+            'component' => 'k-error-view',
+            'props'     => [
+                'error'  => $message,
+                'layout' => Panel::hasAccess($kirby->user()) ? 'inside' : 'outside'
+            ],
+            'title' => 'Error'
+        ];
+    }
+
+    /**
+     * Creates $fiber response array
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @param string $component
+     * @param array $data
+     * @return array
+     */
+    public static function fiber(App $kirby, string $component, array $data = []): array
+    {
+        // get all data for the request
+        $data = static::data($kirby, $data);
+
+        // filter data, if it's a partial request
+        $data = static::partial($kirby, $component, $data);
+
+        // resolve lazy data entries
+        return A::apply($data);
+    }
+
+    /**
+     * Returns the file object for the given
+     * parent path and filename
+     *
+     * @param string|null $path Path to file's parent model
+     * @param string $filename Filename
+     * @return \Kirby\Cms\File|null
+     * @throws \Kirby\Exception\NotFoundException if the file cannot be found
+     */
+    public static function file(string $path = null, string $filename)
+    {
+        $filename = urldecode($filename);
+        $file     = static::parent($path)->file($filename);
+
+        if ($file && $file->isReadable() === true) {
+            return $file;
+        }
+
+        throw new NotFoundException([
+            'key'  => 'file.notFound',
+            'data' => [
+                'filename' => $filename
+            ]
+        ]);
+    }
+
+    /**
+     * Check for access permissions
+     *
+     * @param \Kirby\Cms\User|null $user
+     * @param string|null $areaId
+     * @return bool
+     */
+    public static function firewall(?User $user = null, ?string $areaId = null): bool
+    {
+        // a user has to be logged in
+        if ($user === null) {
+            throw new PermissionException(t('error.access.panel'));
```

Comment:
The translation keys are handled by the permission classes, so this should be:

```suggestion
            throw new PermissionException(['key' => 'access.panel']);
```

The same applies to other uses below.